vhpidirect/quickstart/sharedvar/shrecord/caux.c

```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+
+static const char HDL_LOGIC_CHAR[] = { 'U', 'X', '0', '1', 'Z', 'W', 'L', 'H', '-'};
```

Comment:
At this point of the guide, logic vectors are not introduced yet. There are multiple options:

- Leave it here, but add a note to the docs making it explicit that HDL_LOGIC_CHAR and HDL_LOGIC_STATES are explained in arrays/logicvector.
- Don't use std_logic types here, but stick to constrained strings, character, integer, real, time, etc.
- Move this as a subexample of logicvector. So, instead of "proposing" the readers to jump forward from the quick start guide, we suggest them to jump back from the advanced example.
  - If we did this, we might want to consider renaming 'arrays' to 'arrays and structs'.

Reply:
> Move this as a subexample of logicvector.
 
I think that it should come after the arrays section.. But what we are really getting around to is that "arrays" is the start of explaining sharing more complicated variables. I think ultimately we will have a section on strings too, maybe move the logic-vector to enums, and then have records. So I'd starting to feel like arrays should be a under sharedvar, and these other sections too.

Thoughts?

I think for right now, we can make that reorganisation as another pull yeah? And I will just clean up the example. And just go for adding a note that the logicvector example explains things further.